imports/plugins/core/graphql/server/schemas/cart.graphql

```diff
@@ -0,0 +1,346 @@
+  mode: CartReconciliationMode
+
+  "An optional string identifying the mutation call, which will be returned in the response payload"
+  clientMutationId: String
+}
+
+
+####################
+# Payloads
+# These types are used as return values for mutation calls
+####################
+
+"The payload returned from the `createCart` mutation call"
+type CreateCartPayload {
+  "The modified cart"
+  cart: Cart!
+
+  "The same string you sent with the mutation params, for matching mutation calls with their responses"
+  clientMutationId: String
+}
+
+
+"The payload returned from the `addCartItems` mutation call"
+type AddCartItemsPayload {
+  "The modified cart"
+  cart: Cart!
+
+  "The same string you sent with the mutation params, for matching mutation calls with their responses"
+  clientMutationId: String
+}
+
+
+"The payload returned from the `removeCartItems` mutation call"
+type RemoveCartItemsPayload {
+  "The modified cart"
+  cart: Cart!
+
+  "The same string you sent with the mutation params, for matching mutation calls with their responses"
+  clientMutationId: String
+}
+
+
+"The payload returned from the `updateCartItems` mutation call"
+type UpdateCartItemsPayload {
+  "The modified cart"
+  cart: Cart!
+
+  "The same string you sent with the mutation params, for matching mutation calls with their responses"
+  clientMutationId: String
+}
+
+
+"The payload returned from the `reconcileCart` mutation call"
+type ReconcileCartPayload {
+  "The modified cart"
+  cart: Cart!
+
+  "The same string you sent with the mutation params, for matching mutation calls with their responses"
+  clientMutationId: String
+}
+
+####################
+# Mutations
+####################
+
+extend type Mutation {
+  # Enforce that at least one item is provided so we don't create empty carts.
+  #
+  # IMPORTANT: CartID must be treated as (or more) securely as a session ID. We
+  # need to prevent brute-force cart id collisions.
+  #
+  #   See: https://www.owasp.org/index.php/Session_Management_Cheat_Sheet
+  #
+  #   * ID value must provide at least 64 bits of entropy (if a good PRNG is
+  #     used, this value is estimated to be half the length of the session ID).
+  #   * If a session ID with an entropy of 64 bits is used, it will take an
+  #     attacker at least 292 years to successfully guess a valid session ID,
+  #     assuming the attacker can try 10,000 guesses per second with 100,000
+  #     valid simultaneous sessions available in the web application [2].
+  #
+  "Create a new cart."
+  createCart(input: CreateCartInput!): CreateCartPayload!
+
+  # TODO: Consider returning the connection for the following mutations to
+  # resolve questions of which type should be returned.
+
+  # Add an item to cart.
+  # If the item already already exists in the cart, then increment quantity.
+  # TODO: Consider returning connection?
```

Comment:
@ticean Do we want to figure out this idea of returning a connection here now? Or keep it in mind for later?

Reply:
We can keep it in mind for later since we're not using the pattern now.